android/engine/src/test/java/org/smartregister/fhircore/engine/task/FhirCarePlanGeneratorTest.kt

```diff
@@ -1936,13 +1871,15 @@ class FhirCarePlanGeneratorTest : RobolectricTest() {
     coEvery { defaultRepository.loadResource(Reference(encounter.partOf.reference)) } returns
       immunizationResource
 
-    val updatedTask = runBlocking { groupTask.updateDependentTaskDueDate(defaultRepository) }
+    val updatedTask = runBlocking {
+      groupTask.updateDependentTaskDueDate(defaultRepository, fhirEngine)
+    }
     assertEquals(groupTask, updatedTask)
   }
   @Test
   fun `updateDependentTaskDueDate sets executionPeriod start correctly`() {
```

Comment:
this test is still failing, I'm probably missing a condition needed for find the dependent task

Reply:
Okay will check why.

Reply:
this is now failing b/c of sqlite connection error, in the past this has meant a missing mock for the engine, since I wrapped it in the spyk this make sense, we might need to mock the specific call, we can try passing a pure mock just here to figure out where the problem is, then add the mock, then remove the pure mock and it should pass w/the spyk

Reply:
ooo that was unrelated this goes away in the CI too